Assets/Scripts/Ecosystem/Creature.cs

```diff
@@ -0,0 +1,371 @@
+
+        float angle = Vector3.Angle(creatureToPlanetCenter, posToPlanetCenter);
+        return angle < consumeRadius;
+    }
+
+    private void Rotate()
+    {
+        Vector3 direction = destination - transform.position;
+
+        RaycastHit hit;
+        
+        // Cast a ray down to get the normal of the terrain
+        if (Physics.Linecast(transform.position, planet.transform.position, out hit))
+        {
+            Vector3 newUp = hit.normal;
+            Vector3 oldForward = transform.forward;
+
+            Vector3 newRight = Vector3.Cross(newUp, oldForward);
+            Vector3 newForward = Vector3.Cross(newRight, newUp);
+            
+            // Caluculate angle between the creature and destination
+            float angle = Vector3.SignedAngle(newForward, direction, newUp);
+
+            // Set rotation of creature to the normal of hit
+            transform.rotation = Quaternion.Lerp(transform.rotation, Quaternion.LookRotation(newForward, newUp), Time.fixedDeltaTime * 10f);
+
+            // Rotate the creature to the direction it is walking
+            transform.Rotate(new (0, angle * Time.fixedDeltaTime, 0));
+            
+        }
+    }
+    private void AttractToPlanet()
+    {
+        float gravity = -9.82f; // May want different gravity in the future
+        Vector3 targetDirection = (transform.position - planet.transform.position).normalized;
+
+        rigidbody.AddForce(targetDirection * gravity);
+    }
+
+    private Vector3 GetRandomPoint()
+    {
+        Quaternion rotation = Quaternion.FromToRotation(Vector3.forward, transform.position);
+        Vector3 randomPoint;
+        int tries = 0;
+
+        // Get a random point on the planet, if it fails try again
+        do
+        {
+            randomPoint = transform.position + rotation * Random.insideUnitCircle * detectionRadius * 2;
+            tries++;
+
+            // At 100 tries, just walk back the way you came
+            if (tries > 100)
+            {
+                return transform.position - transform.forward * 4f;
+            }
+        } while (Vector3.Distance(randomPoint, planet.transform.position) < Mathf.Abs(planet.waterDiameter) / 2);
+
+        return randomPoint;
+    }
+
+    // Interacts with a resource and plays eat animation
+    private void InteractWithResourceAction(GameObject resource, bool destroy)
+    {
+        currentState = CreatureState.PerformingAction;
+        animator.SetBool("Walk", false);
+        animator.SetBool("Eat", true);
+        StartCoroutine(InteractWithResource(resource, destroy));
+    }
+
+    private IEnumerator InteractWithResource(GameObject resource, bool destroy)
+    {
+        // Animation clip length
+        float clipLength = animator.GetCurrentAnimatorStateInfo(0).length;
+
+        // Wait for 3 seconds
+        yield return new WaitForSeconds(clipLength);
+        
+        if (destroy) Destroy(resource);
```

Comment:
I can see that you are destroying the food objects but there is no code for having the food grow back. I would like to see that the food can come back if you want the chickens to actually eat the food. But I would suggest just deactivating the object they are eating instead of destroying it, that way we can just activate it again after a certain time and we don't have to do any new creation of objects. We can just reuse the ones we already have